### PR TITLE
ARROW-3697: [Ruby] Add schema#[]

### PR DESCRIPTION
**Pull request by shiro615**

### CODE REVIEW

**Review by kou — CHANGES_REQUESTED** [review]
Sorry. I forgot that `Schema#fields` exists...

File: ruby/red-arrow/lib/arrow/schema.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+module Arrow
+  class Schema
+    include Enumerable
+
+    alias_method :[], :get_field_by_name
```
[the remark this answers is not in the document]
Comment (shiro615) [reply ruby/red-arrow/lib/arrow/schema.rb]: I've extended it.

File: ruby/red-arrow/lib/arrow/schema.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    def each(&block)
+      return to_enum(__method__) unless block_given?
+
+      fields.each(&block)
+    end
```
[the remark this answers is not in the document]
Comment (shiro615) [reply ruby/red-arrow/lib/arrow/schema.rb]: I see! I've removed it.

File: ruby/red-arrow/test/test-schema.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  def setup
+    @count_field = Arrow::Field.new("count", :uint32)
+    @visible_field = Arrow::Field.new("visible", :boolean)
+    @schema = Arrow::Schema.new([@count_field, @visible_field])
+  end
+
+  test("#[]") do
+    assert_equal([@count_field, @visible_field],
+                 [@schema[:count], @schema[:visible]])
+  end
```
[the remark this answers is not in the document]
Comment (shiro615) [reply ruby/red-arrow/test/test-schema.rb]: I've added `test("[Symbol]")` and `test("[Integer]")`.

**Review by kou — APPROVED** [review]
Thanks!
I've added some improvements, removing redundant `#to_s` call and adding the invalid argument to error message.
I'll merge this when CI is passed.

(I reconsidered `Schema#each`. It's not so bad. So we may add it later when someone provides useful use-case.)

**Comment by kou** [comment]
I've merged.

**Pull request closed without merging**